src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodCallTranslator.cs

```diff
@@ -0,0 +1,401 @@
+                return VersionAtLeast(9, 3)
+                    ? new SqlFunctionExpression("array_replace", expression.Method.ReturnType, expression.Arguments.Skip(1).Take(3))
+                    : throw VersionNotSupportedException(nameof(NpgsqlArrayExtensions.ArrayReplace), 9, 3);
+
+            case nameof(NpgsqlArrayExtensions.ArrayToString):
+                return VersionAtLeast(9, 1)
+                    ? new SqlFunctionExpression("array_to_string", expression.Method.ReturnType, expression.Arguments.Skip(1).Take(3))
+                    : VersionAtLeast(7, 4)
+                        ? new SqlFunctionExpression("array_to_string", expression.Method.ReturnType, expression.Arguments.Skip(1).Take(2))
+                        : throw VersionNotSupportedException(nameof(NpgsqlArrayExtensions.ArrayToString), 7, 4);
+
+            case nameof(NpgsqlArrayExtensions.StringToArray):
+                return VersionAtLeast(9, 1)
+                    ? new SqlFunctionExpression("string_to_array", expression.Method.ReturnType, expression.Arguments.Skip(1).Take(3))
+                    : VersionAtLeast(7, 4)
+                        ? new SqlFunctionExpression("string_to_array", expression.Method.ReturnType, expression.Arguments.Skip(1).Take(2))
+                        : throw VersionNotSupportedException(nameof(NpgsqlArrayExtensions.StringToArray), 7, 4);
+
+            case nameof(NpgsqlArrayExtensions.StringToList):
+                return VersionAtLeast(9, 1)
+                    ? new SqlFunctionExpression("string_to_array", expression.Method.ReturnType, expression.Arguments.Skip(1).Take(3))
+                    : VersionAtLeast(7, 4)
+                        ? new SqlFunctionExpression("string_to_array", expression.Method.ReturnType, expression.Arguments.Skip(1).Take(2))
+                        : throw VersionNotSupportedException(nameof(NpgsqlArrayExtensions.StringToList), 7, 4);
+
+            default:
+                return null;
+            }
+
+            NotSupportedException VersionNotSupportedException(string name, int major, int minor)
+                => new NotSupportedException($"{nameof(NpgsqlArrayExtensions)}.{name} is not supported before PostgreSQL {major}.{minor}.");
+        }
+
+        /// <summary>
+        /// Translates static methods defined on <see cref="Array"/>.
+        /// </summary>
+        /// <param name="expression">The expression to translate.</param>
+        /// <returns>
+        /// A translated expression or null if no translation is supported.
+        /// </returns>
+        [CanBeNull]
+        Expression ArrayStaticHandler([NotNull] MethodCallExpression expression)
+        {
+            if (!expression.Method.IsStatic || expression.Method.DeclaringType != typeof(Array))
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case nameof(Array.IndexOf) when VersionAtLeast(9, 5) && expression.Method.GetParameters().Length <= 3:
```

Comment:
From what I can see, I think that `Array.IndexOf` is only intended to target one-dimensional arrays:

```xml
/// <exception cref="T:System.RankException"><paramref name="array">array</paramref> is multidimensional.</exception>
```

If someone passes a multidimensional array (and the compiler allows it), is it our responsibility to leave it to throw on the client? Or should we let it pass and allow PostgreSQL to throw?

Reply:
Yeah, it's true that this is a question of where to throw... `Array.IndexOf()` only accepts multidimensional parameters (and then throws) because they're the same type statically (`Array`). I'd rather we didn't translate and allowed the client to throw.